src/node/txreconciliation.cpp

```diff
@@ -193,6 +210,104 @@ class TxReconciliationTracker::Impl
         LOCK(m_txreconciliation_mutex);
         return IsPeerRegistered(peer_id);
     }
+
+    std::vector<NodeId> GetFanoutTargets(CSipHasher& deterministic_randomizer_with_wtxid,
+                                         bool we_initiate, double limit) const EXCLUSIVE_LOCKS_REQUIRED(m_txreconciliation_mutex)
+    {
+        // The algorithm works as follows. We iterate through the peers (of a given direction)
+        // hashing them with the given wtxid, and sort them by this hash.
+        // We then consider top `limit` peers to be low-fanout flood targets.
+        // The randomness should be seeded with wtxid to return consistent results for every call.
+
+        // To handle fractional values, we add one peer optimistically and then probabilistically
+        // drop it later.
+        double integer_part;
+        double fractional_peer = std::modf(limit, &integer_part);
+        const size_t targets = size_t(integer_part) + 1;
+        const bool drop_peer_if_extra = deterministic_randomizer_with_wtxid.Finalize() > fractional_peer * double(UINT64_MAX);
```

Comment:
Do we have to first add and then (maybe) drop a peer here (instead of determining how many peers we want at the beginning, and then getting as many peers as we can up the desired number).